app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Dashboard.java

```diff
@@ -0,0 +1,165 @@
+import com.aspirebudgetingmobile.aspirebudgeting.models.DashboardCardsModel;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.ObjectFactory;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.SessionConfig;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.SheetsManager;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Dashboard extends Fragment {
+
+    private static final String TAG = "DASHBOARD";
+
+    private View view;
+    private Context context;
+    private SheetsManager sheetsManager;
+    private ObjectFactory objectFactory = ObjectFactory.getInstance();
+    private SessionConfig sessionConfig;
+    private String sheetID = "";
+    private List<List<Object>> fetchedData = null;
+    private JSONArray parsedData = new JSONArray();
+    private List<Object> fakeBuffer = new ArrayList<>();
+
+    private List<DashboardCardsModel> list = new ArrayList<>();
+    private RecyclerView recyclerView;
+    private DashboardCardsAdapter adapter;
+
+    LinearLayout loadingLayout;
+
+    public Dashboard() {}
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        view = inflater.inflate(R.layout.fragment_dashboard, container, false);
+        context = view.getContext();
+
+        // INITIALIZE ALL THE UTIL CLASSES
+        sessionConfig = new SessionConfig(context);
+        sheetsManager = objectFactory.getSheetsManager();
+
+        // FETCH THE SHEET ID FROM SESSION
+        sheetID = sessionConfig.getSheetId();
+
+        // FETCH ID OF VIEWS IN LAYOUT
+        loadingLayout = view.findViewById(R.id.loadingLayout);
+
+        recyclerView = view.findViewById(R.id.dashboardCardsRecyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(context, RecyclerView.VERTICAL, false));
+        adapter = new DashboardCardsAdapter(context, list);
+
+        // CREATED A FAKE LIST ITEM TO INCLUDE THE LAST ITEM IN MAIN DATA
+        fakeBuffer.add("FakeBuffer");
+
+        // FETCH CATEGORIES AND GROUPS TO DISPLAY
+        getGroupToDisplay();
+
+        return view;
+    }
+
+    @SuppressLint("StaticFieldLeak")
+    private void getGroupToDisplay() {
+        new AsyncTask<Void, Void, Void>() {
+
+            @Override
+            protected Void doInBackground(Void... voids) {
+
+                fetchedData = sheetsManager.fetchCategoriesAndGroups(context, sheetID);
+                fetchedData.add(fakeBuffer);
+                return null;
+            }
+
+            @Override
+            protected void onPostExecute(Void aVoid) {
+                super.onPostExecute(aVoid);
+
+                parseFetchedData();
+            }
+        }.execute();
+    }
+
+    private void parseFetchedData() {
```

Comment:
A View should not be responsible for parsing data. A view should only be responsible for displaying the data. Fetching data, parsing data, etc should be handled by Facility classes.

Reply:
Sure, I will shift it to an util class